### ISSUE TITLE
API: Missing fields in the account array

### ISSUE THREAD

**Issue by annando** [friendica/friendica#7968]
The following fields in the account array are published by Mastodon but currently not by Friendica:
* discoverable (boolean) - should be "contact" field "unsearchable"
* group (boolean) - most likely this is "true" if this is a "group" (forum) account.
* last_status_at (date) - Should be fetched from both the personal and the public contact entry from the field "last-item".

Concerning the last field: In the case of a new contact (incoming contact request) or when this is just a follower and we aren't follow this account, then this field will not contain the correct data. So we should always fetch the value from the user's contact entry and the public contact array and took the newest one.

**Comment by MrPetovan** [comment]
I should have known the Mastodon API documentation was too good to be entirely accurate.

**Comment by annando** [comment]
The documentation is missing several things. I guess we have to look directly in their code to see what is missing as well.

**Comment by annando** [comment]
See here for their real endpoints: https://github.com/tootsuite/mastodon/tree/master/spec/controllers/api/v1

**Comment by MrPetovan** [comment]
I assume the Account knowledge is somewhere in there but I really am unfamiliar with Ruby code: https://github.com/tootsuite/mastodon/blob/902c6bed5aab12c2e97af3452f122247777226af/app/models/account.rb

**Comment by MrPetovan** [comment]
And some fields come from here: https://github.com/tootsuite/mastodon/blob/62f60e86c281ae1cd0356a7630dbb76069e2ffde/app/models/account_stat.rb

I'd be better off just using the Mastodon API so see what is returned.

**Comment by annando** [comment]
The `id` field should be taken from the public contact, since it is confusing when you receive different id values for the same account via different endpoints: https://github.com/friendica/friendica/issues/7887#issuecomment-568177060

**Comment by MrPetovan** [comment]
I personally think it's wrong from Mastodon to return a naked `Account` entity when calling the `follow_requests` endpoint even though they do have a `FollowRequest` entity as well. Our follow requests have additional data for example, so it isn't interchangeable with a contact entity.

So while implementing the `follow_requests` endpoint I kept the return value as Account but the ids are for follow requests. This way the `follow_requests` endpoints are consistent with each other. The same is true with Mastodon's endpoints, you can use the `follow_requests` entity ids in the `follow_requests/:id/approve` and `follow_requests/:id/reject` endpoints, but I don't think you should assume any account Id can also be used in those endpoints.

In short Mastodon uses accounts IDs in both `/accounts` endpoints and `/follow_requests` endpoints, we use different IDs but if clients take `/accounts` ids and `/follow_requests` ids independently both will work the same.

**Comment by MrPetovan** [comment]
What do you think, @stom79, it still time to align ids with Masodon's. What would you prefer?

**Comment by stom79** [comment]
@MrPetovan Initially, I didn't get it because I kept the same logic as the Mastodon API. But, I agree that ids for the follow requests endpoint should not be related to an account. Everything, is working fine on my end with follow requests ids, so I would prefer to keep that like it is.
